sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -112,6 +117,17 @@ private static DoFnSignature parseSignature(Class<? extends DoFn<?, ?>> fnClass)
         findAnnotatedMethod(errors, DoFn.GetRestrictionCoder.class, fnClass, false);
     Method newTrackerMethod = findAnnotatedMethod(errors, DoFn.NewTracker.class, fnClass, false);
 
+    Collection<Method> onTimerMethods =
+        declaredMethodsWithAnnotation(DoFn.OnTimer.class, fnClass, DoFn.class);
+
+    HashMap<String, DoFnSignature.OnTimerMethod> onTimerMethodMap =
+            Maps.newHashMapWithExpectedSize(onTimerMethods.size());
```

Comment:
I prefer to make anything depending on order explode. Is there something in particular you have in  mind?

I could imagine it being useful to be able to give error message about code in the order the user wrote it in the file, but that's a different argument that doesn't apply to today's logic.